### PR TITLE
SDK-1562 R8 minification bug fix

### PR DESCRIPTION
**Pull request by jhaven-stytch**
Since the upgrade to AGP 8, R8 fullMode is enabled, which was stripping out some class definitions when minification is enabled.

Linear Ticket: [SDK-1562](https://linear.app/stytch/issue/SDK-1562)

## Changes:

1. Bump retrofit and moshi versions to incorporate new proguard rules
2. Annotate our requests/responses with @Keep to prevent them from being stripped (there's probably a less-verbose/manual way to do this in our proguard files, but I spent 2 hours trying to figure it out with no success :( )
3. Force minify in workbench apps for testing purposes
4. Bump version

## Notes:

- I also published this version to my local maven repo and tested in an "external" app and verified that its working there

## Checklist:
- [x] I have verified that this change works in the relevant demo app, or N/A
- [x] I have added or updated any tests relevant to this change, or N/A
- [x] I have updated any relevant README files for this change, or N/A

### CODE REVIEW

File: sdk/proguard-rules.pro [hunk 1]
```diff
@@ -28,3 +28,4 @@
 -keep class androidx.credentials.playservices.** {
   *;
 }
+-dontwarn java.lang.invoke.StringConcatFactory
```
Comment (alex-stytch) [inline sdk/proguard-rules.pro:31]: should there be a new line?

**Review by alex-stytch — APPROVED** [review]
Do you know why the UI tests are failing?

File: sdk/proguard-rules.pro (hunk 1 above)
Comment (jhaven-stytch) [reply sdk/proguard-rules.pro]: Added

**Comment by jhaven-stytch** [comment]
> Do you know why the UI tests are failing?

They appear to be a little flaky in the AVD. I'm seeing random low-level errors like `E0403 18:34:28.985345 f47a000 FrameBuffer.cpp:3790] Failed to find ColorBuffer:333`, which may be contributing? Each time they fail, it's a different test, so I know it's not a code issue/actually failing test

**Comment by jhaven-stytch** [comment]
Tests passed after clearing out the old caches.

**Pull request merged**